### PR TITLE
Containerize the application with multistage builds

### LINKED ISSUE #3: feat: dockerize the code for faster and reliable deployment
### Tasks
- There should be a d**ocker-compose** file and/or **dockerfile** to generate builds for faster and reliable deployments.

### PR DESCRIPTION
**Pull request by h3110Fr13nd**
closes #3 
- Used node:20-alpine for small image size, and node version 20 to be consistent with development
- Created a Dockerfile and docker-compose.yaml file
- build by typing docker compose up --build, run by docker compose up
- requires .env in folder for now.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+services:
+  gitslackbot:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - 3002:3002
+      - 3004:3004
+    env_file:
+      - .env
```
Comment (HarshDobariya79) [inline docker-compose.yaml]: we'll also be needing a persistent volume for `logs/`.

**Review by HarshDobariya79 — CHANGES_REQUESTED** [review]
Nice work with **multi-stage** build with `.dockerignore`. I've added some minor changes we might need. Please have a look.

File: docker-compose.yaml
Comment (HarshDobariya79) [inline docker-compose.yaml:1]: Can you also specify the restart option to make sure the bot is always up.

Comment (h3110Fr13nd) [reply docker-compose.yaml:1]: I don't know how restart policy works. in compose exactly. Yet have kept restart policy to always.
Please change it and tell me if that's not correct.
```docker-compose
version: '3'
services:
  gitslackbot:
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - 3002:3002
      - 3004:3004
    env_file:
      - .env
    volumes:
      - logs:/app/logs
    restart: always
volumes:
  logs:
```

Comment (HarshDobariya79) [reply docker-compose.yaml:1]: yes `always` works quite well.

**Review by HarshDobariya79 — APPROVED** [review]
Thank you for implementing the requested changes.

**Pull request merged**